gpu4pyscf/scf/uhf.py

```diff
@@ -157,26 +155,54 @@ def eig(self, fock, s):
         e_b, c_b = self._eigh(fock[1], s)
         return cupy.array((e_a,e_b)), cupy.array((c_a,c_b))
 
-    def get_veff(self, mol=None, dm=None, dm_last=None, vhf_last=0, hermi=1):
+    def get_veff(self, mol=None, dm=None, dm_last=0, vhf_last=0, hermi=1):
         if mol is None: mol = self.mol
         if dm is None: dm = self.make_rdm1()
-        if getattr(dm, 'ndim', 0) == 2:
+
+        if isinstance(dm, cupy.ndarray) and dm.ndim == 2:
             dm = cupy.asarray((dm*.5,dm*.5))
 
-        if dm_last is None or not self.direct_scf:
-            vj, vk = self.get_jk(mol, dm, hermi)
+        if self._eri is not None or not self.direct_scf:
+            vj, vk = self.get_jk(mol, cupy.asarray(dm), hermi)
             vhf = vj[0] + vj[1] - vk
         else:
             ddm = cupy.asarray(dm) - cupy.asarray(dm_last)
             vj, vk = self.get_jk(mol, ddm, hermi)
             vhf = vj[0] + vj[1] - vk
-            vhf += vhf_last
+            vhf += cupy.asarray(vhf_last)
         return vhf
 
+    def scf(self, dm0=None, **kwargs):
```

Comment:
reuse the same method in hf.py

Reply:
Should I make the scf in the rhf.py out of the RHF class, and reuse it in the UHF class in the uhf.py?

Reply:
> reuse the same method in hf.py
Should I make the scf in the rhf.py out of the RHF class, and reuse it in the UHF class in the uhf.py?